src/init/welcome.ts

```diff
@@ -9,21 +9,22 @@ import { activationCommand } from '../commands';
 const i18nManager = new LanguageResourceManager(vscode.env.language);
 
 export function manageWelcomeMessage(settingsManager: ISettingsManager): void {
-  const themeName = getConfig().inspect(constants.vscode.iconThemeSetting).globalValue;
-  if (!settingsManager.getState().welcomeShown || themeName !== constants.extensionName) {
-    showWelcomeMessage();
```

Comment:
Consider the case where `welcomeShown` for any reason is set to `false` but `vscode-icons` is already set as the `theme` in settings (which means that it's already activated). I still believe the proper way is to switch the condition from disjunction to conjunction.

Reply:
You've got a point!